ui/app/AppLayouts/Onboarding2/components/LoginKeycardBox.qml

```diff
@@ -37,6 +37,12 @@ Control {
         pinInputField.forceFocus()
     }
 
+    function onWrongPin() {
+        d.wrongPin = true
+        pinInputField.statesInitialization()
+        pinInputField.forceFocus()
+    }
+
```

Comment:
The method name should be sth like "markAsWrongPin`, `onXXX` is specific for signal handlers and should be avoided for functions imo.

Anyways, it would be much cleaner and consistent with other async cases to have the UI driven by some enum representing login state, like (Idle, InProgress, Error, Success).